Panic Due to Index Out of Range

## Summary

When I tried to run the library with the file I provided below, I noticed a crash. I'm sharing the necessary PoC and file with you.

## PoC

```golang
package main

import (
       "github.com/nlpodyssey/gopickle/pickle"	
)

func main() {
	_, _ = pickle.Load("./crashers/fe1fed2bfa2c02c11be163a4e4f6400bcfd67a1e") 
}

```

## Crash Log

```
panic: runtime error: index out of range [7] with length 7

goroutine 1 [running]:
github.com/nlpodyssey/gopickle/pickle.loadDict(0xc00018e000)
        /home/ubuntu/go/pkg/mod/github.com/nlpodyssey/gopickle@v0.3.0/pickle/pickle.go:953 +0x27e
github.com/nlpodyssey/gopickle/pickle.(*Unpickler).Load(0xc00018e000)
        /home/ubuntu/go/pkg/mod/github.com/nlpodyssey/gopickle@v0.3.0/pickle/pickle.go:102 +0xe6
github.com/nlpodyssey/gopickle/pickle.Load({0x4df614?, 0xc0000061c0?})
        /home/ubuntu/go/pkg/mod/github.com/nlpodyssey/gopickle@v0.3.0/pickle/pickle.go:31 +0x318
main.main()
        /home/ubuntu/targets/go
pickle_fuzz/maintest.go:8 +0x1f
exit status 2
```


[crashpickle.zip](https://github.com/nlpodyssey/gopickle/files/14695419/crashpickle.zip)

--- a/pickle/pickle.go
+++ b/pickle/pickle.go
@@ -948,8 +948,8 @@ func loadDict(u *Unpickler) error {
 		return err
 	}
 	d := types.NewDict()
-	itemsLen := len(items)
-	for i := 0; i < itemsLen; i += 2 {
+	n := len(items)
+	for i := 0; i < n-1; i += 2 {
 		d.Set(items[i], items[i+1])
 	}
 	u.append(d)

--- a/pickle/pickle_test.go
+++ b/pickle/pickle_test.go
@@ -6,6 +6,7 @@ package pickle
 
 import (
 	"fmt"
+	"io"
 	"math/big"
 	"reflect"
 	"strings"
@@ -751,6 +752,20 @@ func TestP4Carray(t *testing.T) {
 				t.Fatalf("got=%v, want=%v", got, tc.want)
 			}
 		})
+	}
+}
+
+func TestIssue16(t *testing.T) {
+	var (
+		pkl  = "\x28\x88\x88\x88\x88\x88\x88\x88\x64"
+		want = io.EOF
+	)
+	_, err := Loads(pkl)
+	if err == nil {
+		t.Fatalf("expected an error")
+	}
+	if got, want := err.Error(), want.Error(); got != want {
+		t.Fatalf("invalid error:\ngot= %q\nwant=%q", got, want)
 	}
 }
 

